### PR TITLE
[2.13.x] DDF-4130 Fixed searching an OpenSearchSource by a keyword that contains a space

### PR DESCRIPTION
**Pull request by emmberk**
#### What does this PR do?
This PR fixes querying an OpenSearchSource by a keyword that contains a space.
#### Who is reviewing it? 
@peterhuffer @kcover @lamhuy 
#### Select relevant component teams: 
@codice/io 
#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@clockard
#### How should this be tested?

1. Ingest a product that contains a phrase with spaces. (I modified [this test file](https://github.com/codice/ddf/blob/dfbce3ca3636bb80c98e5c2271e917686e5dd8d9/distribution/test/itests/test-itests-common/src/main/resources/metacard1.xml#L22) to have a title of `Metacard 1`.)
2. Create a loopback OpenSearch source.
3. Confirm that the product is returned when you search for it by the phrase. 

#### Any background context you want to provide?
The search phrase was being encoded twice: first with the URLEncoder (the part that is removed in the PR) and second in the `WebClient` after `replaceQueryParam`.
#### What are the relevant tickets?
[DDF-4130](https://codice.atlassian.net/browse/DDF-4130)
#### Checklist:
- [ ] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests
#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/opensearch/catalog-opensearch-source/src/test/groovy/org/codice/ddf/opensearch/source/OpenSearchParserImplSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.apache.cxf.jaxrs.client.WebClient
+import spock.lang.Specification
+
+class OpenSearchParserImplSpec extends Specification {
+
+    private OpenSearchParser openSearchParser = new OpenSearchParserImpl()
+
+    private WebClient webClient = Mock(WebClient)
+
+    // {@link OpenSearchParser#populateContextual(WebClient, Map, List)} tests
```
Comment (peterhuffer) [inline catalog/opensearch/catalog-opensearch-source/src/test/groovy/org/codice/ddf/opensearch/source/OpenSearchParserImplSpec.groovy]: ✏️ Extra comment.

Comment (emmberk) [reply catalog/opensearch/catalog-opensearch-source/src/test/groovy/org/codice/ddf/opensearch/source/OpenSearchParserImplSpec.groovy]: That was copied from the old file for consistency:
https://github.com/codice/ddf/blob/7e5f36cce802e4b7547ff6a3a880db6d1e6d850a/catalog/opensearch/catalog-opensearch-source/src/test/java/org/codice/ddf/opensearch/source/OpenSearchParserImplTest.java#L80

**Comment by lamhuy** [comment]
should there be an update to federation itest as well to cover this case?

**Comment by emmberk** [comment]
@lamhuy I think that an itest is overkill, but I could be persuaded. The problem is pretty contained to opensearch-source. I don't think we generally tests all edge cases of search criteria for every source in the itests.

**Comment by lamhuy** [comment]
i don't have any strong opinion about it. Just want to make sure url encoding/decoding of `space` is properly handled. But that is on CXF to ensure.

**Comment by emmberk** [comment]
build now

**Comment by peterhuffer** [comment]
Succesful hero. Setup a OpenSearch Federated source to Emily's machine and verified I could search the results with spaces in the search keyword. ✅
